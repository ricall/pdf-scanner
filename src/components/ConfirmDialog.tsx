import React from "react";
import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from "@material-ui/core";

export interface ConfirmDialogProps {
    display: boolean;
    onAccept: () => void;
    onReject: () => void;
}

export const ConfirmDialog: React.FC<ConfirmDialogProps> = ({ display, onAccept, onReject }) => (
    <Dialog open={display} onClose={onReject} aria-labelledby="alert-dialog-title"
            aria-describedby="alert-dialog-description">
        <DialogTitle id="alert-dialog-title">Confirm new document?</DialogTitle>
        <DialogContent>
            <DialogContentText id="alert-dialog-description">
                Creating a new document will clear any editing done on the current document.
            </DialogContentText>
        </DialogContent>
        <DialogActions>
            <Button onClick={onReject} color="primary" autoFocus>Disagree</Button>
            <Button onClick={onAccept} color="primary">Agree</Button>
        </DialogActions>
    </Dialog>
);
