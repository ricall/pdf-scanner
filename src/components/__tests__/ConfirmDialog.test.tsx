import React from 'react';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { ConfirmDialog } from "../ConfirmDialog";

describe('ConfirmDialog component', () => {

    it('can be rendered', () => {
        const wrapper = shallow(<ConfirmDialog display={false} onAccept={jest.fn()} onReject={jest.fn()} />);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

});
