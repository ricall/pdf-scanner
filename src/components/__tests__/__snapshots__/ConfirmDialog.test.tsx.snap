// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog component can be rendered 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  onClose={[MockFunction]}
  open={false}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="alert-dialog-title"
  >
    Confirm new document?
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))
      id="alert-dialog-description"
    >
      Creating a new document will clear any editing done on the current document.
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      color="primary"
      onClick={[MockFunction]}
    >
      Disagree
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[MockFunction]}
    >
      Agree
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
